### PR TITLE
Wire channel options to vm

### PR DESCRIPTION
**Pull request by carterhudson**
### Description

Wires up the channel item "more options" menu to perform actions. Also fixes some color issues between day / night modes. 

https://user-images.githubusercontent.com/2574098/103039899-36cb3a80-452f-11eb-801e-8ddf86a48be5.mov

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/ChannelListFragment.kt [hunk 1]
```diff
@@ -58,13 +57,38 @@ class ChannelListFragment : Fragment() {
...
             }
 
-            setDeleteListener {
-                ChatLogger.instance.logD(it.name, "delete me!")
+            setChannelDeleteClickListener { channel ->
+                ChatInfoDeleteChannelDialogFragment
+                    .newInstance()
+                    .apply {
+                        deleteChannelListener = ChatInfoDeleteChannelDialogFragment.ChatInfoDeleteChannelListener {
+                            // should clear history, leave & hide?
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/ChannelListFragment.kt:70]: Did you figure out it? Or is it kind of TODO?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/ChannelListItemAdapter.kt [hunk 2]
```diff
@@ -112,4 +113,13 @@ internal class ChannelListItemAdapter(var style: ChannelListViewStyle) :
...
         bind(position, holder, NOTHING_CHANGED)
     }
+
+    internal fun getChannel(cid: String): Channel {
+        return currentList
+            .asSequence()
+            .filter { it is ChannelListItem.ChannelItem }
+            .map { it as ChannelListItem.ChannelItem }
+            .first { it.channel.cid == cid }
+            .channel
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/ChannelListItemAdapter.kt:123]:
```suggestion
        return currentList
            .asSequence()
            .filterIsInstance<ChannelListItem.ChannelItem>()
            .first { it.channel.cid == cid }
            .channel
```

**Review by ogkuzmin — DISMISSED** [review]

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/ChannelListItemAdapter.kt (hunk 2 above)
Comment (carterhudson) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/ChannelListItemAdapter.kt:123]: I always forget about `filterIsInstance`

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/ChannelListFragment.kt (hunk 1 above)
Comment (carterhudson) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/ChannelListFragment.kt:70]: Sort of a TODO. Not sure what the appropriate order of events is. Backend functionality for leave is still a WIP. Clear history was just sort of me wondering. I think deleting channel might be a little more complicated for group scenarios, and we're still trying to figure out what the "correct" flow is. I was told I should just emulate what Rafal did in ChatInfo, so it only hides for now.

**Review by ogkuzmin — APPROVED** [review]

**Pull request merged**